util/src/main/java/io/kubernetes/client/informer/cache/DeltaFIFO.java

```diff
@@ -0,0 +1,505 @@
+package io.kubernetes.client.informer.cache;
+
+import static io.kubernetes.client.informer.cache.Deltas.DeltaType.Deleted;
+import static io.kubernetes.client.informer.cache.Deltas.DeltaType.Sync;
+
+import io.kubernetes.client.informer.cache.function.KeyFunc;
+import io.kubernetes.client.informer.cache.function.PopProcessFunc;
+import io.kubernetes.client.informer.exception.EmptyDeltasException;
+import io.kubernetes.client.util.common.Collections;
+import java.util.*;
+import java.util.concurrent.locks.Condition;
+import java.util.concurrent.locks.ReadWriteLock;
+import java.util.concurrent.locks.ReentrantReadWriteLock;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+// DeltaFIFO is a java portable of k/client-go's DeltaFIFO
```

Comment:
Isn't DeltaQueue a better name?

Reply:
both naming sounds fine to me. while `DeltaFIFO` keeps the naming from its go code which might be better for code maintenance. wdyt?

[https://github.com/kubernetes/kubernetes/blob/cdbc9d8567c177b8c6505181f755440792b9ff06/staging/src/k8s.io/client-go/tools/cache/delta_fifo.go#L96](https://github.com/kubernetes/kubernetes/blob/cdbc9d8567c177b8c6505181f755440792b9ff06/staging/src/k8s.io/client-go/tools/cache/delta_fifo.go#L96)